File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        L = 0
        R = len(nums) - 1
        if nums[L] < nums[R]:
            return nums[L]
        while nums[L] >= nums[R] and ((R - L) > 1):
            mid = (L + R) // 2
            if nums[mid] > nums[L]:
                L = mid
            else:
                R = mid
        
            
        return nums[R]

File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> hashmap;
        for (int i = 0; i < (int)nums.size(); i++){
            int need = target - nums[i];
            if (hashmap.contains(need)){
                return {hashmap.at(need), i};
            }
            else{
                hashmap[nums[i]] = i;
            }
        }
        
    }
};
